frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -47,14 +49,20 @@ const selectorReducer = (state: State, action: Action): State => {
     // User actions.
 
     case "ADD_PROJECTS": {
-      // a given custom project may already exist in the group so don't trigger a state update for those duplicates
+      let newState = { ...state };
```

Comment:
don't need to copy state until :65

Reply:
will address this in "remove projects" PR as well